File: Ahorn/entities/DreamRefill.jl
module CommunalHelperDreamRefill

using ..Ahorn, Maple

@mapdef Entity "CommunalHelper/DreamRefill" DreamRefill(x::Integer, y::Integer, oneUse::Bool=false)

const placements = Ahorn.PlacementDict(
    "Dream Refill (Communal Helper)" => Ahorn.EntityPlacement(
        DreamRefill,
        "point"
    )
)

const sprite = "objects/CommunalHelper/dreamRefill/idle02"

function Ahorn.selection(entity::DreamRefill)
    x, y = Ahorn.position(entity)
    return Ahorn.getSpriteRectangle(sprite, x, y)
end

Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::DreamRefill) = Ahorn.drawSprite(ctx, sprite, 0, 0)

end

File: Loenn/entities/dream_refill.lua
local utils = require("utils")

local dreamRefill = {}

dreamRefill.name = "CommunalHelper/DreamRefill"
dreamRefill.depth = -100

dreamRefill.placements = {
    {
        name = "normal",
        data = {
            oneUse = false,
            respawnTime = 2.5
        }
    }
}

dreamRefill.texture = "objects/CommunalHelper/dreamRefill/idle02"

function dreamRefill.rectangle(room, entity)
    return utils.rectangle(entity.x - 5, entity.y - 5, 10, 10)
end

return dreamRefill
